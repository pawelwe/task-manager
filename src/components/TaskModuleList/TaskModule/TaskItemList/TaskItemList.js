import React from 'react';
import Task from './TaskItem/Task';
import classes from './TaskItemList.css';

const TaskItemList = ({ tasks, toggleTask, removeTask, moduleId }) => {
  if(tasks.length < 1) {
    return null;
  }

  return (
    <ul className={classes.TaskItemList}>
      {tasks.map(task => {
        return (
          <Task
            key={task.id}
            id={task.id}
            {...task}
            toggleTask={toggleTask}
            removeTask={removeTask}
            moduleId={moduleId}
          />
        );
      })}
    </ul>
  );
};

export default TaskItemList;
