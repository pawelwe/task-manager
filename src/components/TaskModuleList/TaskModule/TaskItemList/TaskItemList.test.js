import React from 'react';
import { shallow } from 'enzyme';
import TaskItemList from './TaskItemList';

describe('TaskList Component', () => {
  const wrapper = shallow(<TaskItemList tasks={[]} />);

  it('Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
