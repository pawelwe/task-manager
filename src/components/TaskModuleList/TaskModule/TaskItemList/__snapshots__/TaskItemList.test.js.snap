// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList Component Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskItemList
    tasks={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
